third-party/re2/re2-src/re2/nfa.cc

```diff
@@ -700,8 +700,8 @@ bool NFA<StrPiece>::Search(const StrPiece& text, const StrPiece& const_context,
     for (int i = 0; i < nsubmatch; i++)
       submatch[i].set_ptr_end(match_[2 * i], match_[2 * i + 1]);
     if (ExtraDebug)
-      fprintf(stderr, "match (%td,%td)\n",
-              match_[0] - btext_, match_[1] - btext_);
+      fprintf(stderr, "match (%td,%ld)\n",
+              (long)(match_[0] - btext_), (long)(match_[1] - btext_));
```

Comment:
I don't think this change is right.  These should be type ptrdiff_t, which is what you get by default when you subtract two pointers.  The %td handles that.  Casting to long can chop off bits (in particular on 64-bit Windows).  I think this part of the code should be left as is.

The previous version of re2 was worse, though.  It did a static_cast<int> which will chop off bits in a lot of environments.